app/validators/sections/section_validator.py

```diff
@@ -110,27 +94,17 @@ def validate_blocks(self, group_id):
 
     def validate_routing(self, schema_element, group):
         if "routing_rules" in schema_element:
-            if any("goto" in rule for rule in schema_element["routing_rules"]):
-                routing_validator = RoutingValidator(
-                    schema_element, group, self.questionnaire_schema
-                )
-            else:
-                routing_validator = NewRoutingValidator(
-                    routing_rules=schema_element["routing_rules"],
-                    group=group,
-                    origin_id=schema_element["id"],
-                    questionnaire_schema=self.questionnaire_schema,
-                )
+            routing_validator = RoutingValidator(
+                routing_rules=schema_element["routing_rules"],
+                group=group,
+                origin_id=schema_element["id"],
+                questionnaire_schema=self.questionnaire_schema,
+            )
             self.errors += routing_validator.validate()
         if "skip_conditions" in schema_element:
```

Comment:
Minor could make use of walrus operator.

Reply:
Made use of the walrus